src/libs/UserUtils.ts

```diff
@@ -83,21 +100,47 @@ function getDefaultAvatar(accountID = -1): React.FC<SvgProps> {
 
     // There are 24 possible default avatars, so we choose which one this user has based
     // on a simple modulo operation of their login number. Note that Avatar count starts at 1.
-    const accountIDHashBucket = ((accountID % CONST.DEFAULT_AVATAR_COUNT) + 1) as AvatarRange;
 
+    // When we create a chat with a new user, an optimistic ID is used, and the avatar is generated based on it.
+    // When the response is received from the backend, the optimistic ID is replaced with the actual user ID.
+    // However, the avatar link returned by the backend still matches the one generated using the optimistic ID.
+    // Therefore, we cannot directly use the actual user ID to retrieve the SVG image number for the avatar.
+    // Instead, we extract it from the avatar link returned by the backend.
+    let accountIDHashBucket: AvatarRange;
+    if (avatarURL) {
+        const match = avatarURL.match(/(?<=default-avatar_)\d+(?=\.)/);
+        const lastDigit = match && parseInt(match[0], 10);
+        accountIDHashBucket = lastDigit as AvatarRange;
+    } else {
+        accountIDHashBucket = ((accountID % CONST.DEFAULT_AVATAR_COUNT) + 1) as AvatarRange;
+    }
     return defaultAvatars[`Avatar${accountIDHashBucket}`];
 }
 
 /**
  * Helper method to return default avatar URL associated with login
  */
-function getDefaultAvatarURL(accountID: string | number = '', isNewDot = false): string {
+function getDefaultAvatarURL(accountID: string | number = '', isNewDot = true): string {
     if (Number(accountID) === CONST.ACCOUNT_ID.CONCIERGE) {
         return CONST.CONCIERGE_ICON_URL;
     }
-
+    // To ensure that the avatar remains unchanged and matches the one returned by the backend,
+    // utilize an optimistic ID generated from the email instead of directly using the user ID.
+    let email; let originAccountID;
+    if (allPersonalDetails?.[accountID]) {
+        if (allPersonalDetails[accountID].login) {
+            email = allPersonalDetails[accountID].login;
+        } else if (allPersonalDetails[accountID].displayName) {
+            email = allPersonalDetails[accountID].displayName;
+        }
+    }
+    if (email) {
+        originAccountID = generateAccountID(email);
```

Comment:
If I understand right, I think this would clearer if it was named `originalOptimisticAccountID`?

Reply:
OK, I have updated it to originalOptimisticAccountID.